rs/http_endpoints/public/src/metrics.rs

```diff
@@ -170,10 +170,10 @@ impl HttpHandlerMetrics {
             ),
             ingress_watcher_wait_for_certification_duration_seconds: metrics_registry.histogram(
                 "replica_http_ingress_watcher_wait_for_certification_duration_seconds",
-                "The duration the call v3 handler waits for subscribing to a message. I.e. `IngressWatcherHandle::subscribe_for_certification()`.",
+                "The duration the call v3 handler waits for a message to complete execution at some height, h, and for h to become certified.",
                 // We have the final bucket at 10s as that is the maximum time the handler waits for certification
-                // 10ms - 10s.
-                add_bucket(10.0, decimal_buckets(-2, 0)),
+                // 0.75s - 1.10s - 1.45s - 1.70s - ... - 10.5s
+                add_bucket(10.0, linear_buckets(0.75, 0.35, 30)),
```

Comment:
I wonder what an optimal bucket distribution would be. Ideally, you would want to have 0.1 steps between 0.5-3s and bigger afterward.

While 30 buckets is a lot is should be fine since it is an important metric.

Reply:
I agree, we should have even smaller widths on the buckets on the lower end where most of the distribution will be.

[Here's](https://grafana.mainnet.dfinity.network/d/ingress-watcher/ingress-watcher?orgId=1&from=1724249163674&to=1724252094747&var-datasource=IC%20Metrics%20%28cluster%20local%29&var-ic=mercury&var-ic_subnet=snjp4-xlbw4-mnbog-ddwy6-6ckfd-2w5a2-eipqo-7l436-pxqkh-l6fuv-vae&var-instance=All&var-period=$__auto_interval_period&viewPanel=6) a link to a few samples from a 13 node subnet in production with the counter canister. These durations will be even longer for large subnets and most other use cases.

We could do 35 buckets with a step size of 0.1s, followed up with 15 more buckets with a step size of 0.5s.
`0.5s - 0.6s - ... - 4s - 4.5s - ... - 12s`.

What do you think?

Reply:
> While 30 buckets is a lot is should be fine since it is an important metric.

Are there any performance costs for the prometheus instance with this many buckets?

Reply:
It is mostly that the prometheus instance will store a lot of data which leads to lower retention.